platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/internal/DefaultJavaFeatureSpec.java

```diff
@@ -66,7 +66,16 @@ public void create() {
             throw new GradleException("Cannot register feature '" + name + "' because multiple JVM components are present.  These components were found: " + componentNames + ".");
         }
 
-        setupConfigurations(sourceSet);
+        JvmFeatureInternal feature = createFeature();
+
+        // TODO: #23495 Investigate the implications of using this class without
+        //       the java plugin applied, and thus no java component present.
+        //       In the long run, all domain objects created by this feature should be
+        //       owned by a component. If we do not add them to the default java component,
+        //       we should be adding them to a user-provided or new component instead.
+        if (!jvmComponents.isEmpty()) {
+            addFeatureToComponent(feature, jvmComponents.iterator().next());
+        }
```

Comment:
❓ This seems like trouble.  Why is the feature adding itself to the first/only JVM component upon creation?  

Shouldn't the java plugin/extension both create the feature and then add it to the component?

Reply:
Lets move this logic up into the extension. The spec for a feature should not be aware of components